activation/activation_errors.go

```diff
@@ -0,0 +1,35 @@
+package activation
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrStopRequested is returned when builder is stopped.
+	ErrStopRequested = errors.New("builder: stop requested")
+	// ErrATXChallengeExpired is returned when atx missed its publication window and needs to be regenerated.
+	ErrATXChallengeExpired = errors.New("builder: atx expired")
+	// ErrPoetServiceUnstable is returned when poet quality of service is low.
+	ErrPoetServiceUnstable = &PoetSvcUnstableError{}
```

Comment:
The reason `ErrPoetServiceUnstable` exists is to be able to call `errors.Is(err, ErrPoetServiceUnstable)` when checking for that specific error.

I don't mind having this variable, but it could be removed and `errors.As` be used instead of `errors.Is`.

Reply:
right. but the only places that check errors.Is() against `ErrPoetServiceUnstable` is in node_service.go and in tests.
for node_service.go, the differentiation of codes.Unavailable and codes.Internal does not make sense.
for tests, i'd prefer more specific errors to be checked, not a category of `ErrPoetServiceUnstable`.
for example, errUnableToRegister, errServiceIdUnavailable...

Reply:
IMO, the main reason to have different error **types** is to be able to react differently to different errors (control flow). If there is no possible error handling, then simple `fmt.Errorf()` is enough - I'm not a fan of creating error types which are then only logged. `ErrPoetServiceUnstable` is handled in one critical place and thus must be identified: https://github.com/spacemeshos/go-spacemesh/blob/721cf5c9816b29ac3f1c2531c66b2d3447b45745/activation/activation.go#L410